controllers/ProposalController.php

```diff
@@ -676,4 +747,109 @@ private function saveEditedComment(Comment $comment, $newCommentContent)
+            $selectedProposal = $this->checkIfProposalExists($id);
+            $displayItems = $this->buildOneProposalDisplayItems($selectedProposal);
+
+            return $this->render('proposal', $displayItems);
+        }
+
+        $approvedProposalsQuery = $this->buildApprovedProposalsQuery();
+        $approvedProposals = $this->buildApprovedProposalsActiveDataProvider($approvedProposalsQuery);
+        $notApprovedProposals = $this->getNotApprovedProposals($approvedProposalsQuery);
+
+        return $this->render('manage-proposals', [
+           'approvedProposals' => $approvedProposals,
+           'notApprovedProposals' => $notApprovedProposals
+        ]);
+    }
+
+    private function buildApprovedProposalsQuery()
+    {
+        return \app\models\Proposal::find()
+            ->select('proposal.*, (SELECT COUNT(*) FROM review WHERE review.proposal_id = proposal.id) as count_reviews')
+            ->where('1 = CASE 
+                WHEN 
+                    ((((SELECT COUNT(*) FROM review 
+                        WHERE review.status = \'approved\' AND review.proposal_id = proposal.id)
+                        /
+                        (SELECT COUNT(*) FROM review
+                        WHERE review.proposal_id = proposal.id))
+                    *100)  
+                    >= ' . yii\helpers\Html::encode(ProposalApprovementSetting::getApprovementPercent()). ') 
+                    THEN 1 
+                ELSE 0 
+                END')
+            ->andWhere(['proposal.status' => \app\models\Proposal::STATUS_PENDING])
+            ->andWhere('1 = CASE
+                WHEN
+                    ((SELECT COUNT(*) FROM review
+                    WHERE review.proposal_id = proposal.id)
+                    >= ' . yii\helpers\Html::encode(ProposalApprovementSetting::getRequiredNumberOfReview()) . ')
+                    THEN 1
+                ELSE 0
+                END');
+    }
+
+    private function buildApprovedProposalsActiveDataProvider(yii\db\ActiveQuery $approvedProposalsQuery)
+    {
+        return new ActiveDataProvider([
+            'query' => $approvedProposalsQuery,
+            'pagination' => [
+                'pageSize' => 20,
+                'defaultPageSize' => 20
+            ],
+            'sort' => [
+                'sortParam' => 'approvedSort',
+                'attributes' => ['count_reviews', 'date', 'title'],
+                'defaultOrder' => [
+                    'count_reviews' => SORT_DESC,
+                    'date' => SORT_DESC,
+                    'title' => SORT_ASC
+                ]
+            ]
+        ]);
+    }
+
+    private function getNotApprovedProposals(yii\db\ActiveQuery$approvedProposalsQuery)
+    {
+        $approvedProposals = $approvedProposalsQuery->all();
+        $approvedProposalsId = array();
+
+        foreach ($approvedProposals as $approvedProposal) {
+                array_push($approvedProposalsId, $approvedProposal->id);
+        }
```

Comment:
This is heavy : You get all approved proposals, and then you build an array with only their ID. Why not just retrieve approved proposals IDs with one database query ?

Reply:
The process and I mean the query to return the approved proposals is quite heavy too. It's more easier to understand the code with this approach. 

`    private function getNotApprovedProposals(yii\db\ActiveQuery$approvedProposalsQuery)
    {
        $approvedProposals = $approvedProposalsQuery->all();
        $approvedProposalsId = array();

        foreach ($approvedProposals as $approvedProposal) {
                array_push($approvedProposalsId, $approvedProposal->id);
        }`

is lighter than

`\app\models\Proposal::find()
            ->select('proposal.*')
            ->where('1 = CASE 
                WHEN 
                    ((((SELECT COUNT(*) FROM review 
                        WHERE review.status = \'approved\' AND review.proposal_id = proposal.id)
                        /
                        (SELECT COUNT(*) FROM review
                        WHERE review.proposal_id = proposal.id))
                    *100)  
                    >= ' . yii\helpers\Html::encode(ProposalApprovementSetting::getApprovementPercent()). ') 
                   THEN 1 
                ELSE 0 
                END')
            ->andWhere(['proposal.status' => \app\models\Proposal::STATUS_PENDING])
            ->andWhere('1 = CASE
                WHEN
                    ((SELECT COUNT(*) FROM review
                    WHERE review.proposal_id = proposal.id)
                    >= ' . yii\helpers\Html::encode(ProposalApprovementSetting::getRequiredNumberOfReview()) . ')
                    THEN 1
                ELSE 0
                END');`

Reply:
(It seems that i can't handle indent). It's true that we can also refactorize this query into a method then it will be lighter.